torch/fx/passes/utils/fuser_utils.py

```diff
@@ -50,23 +50,36 @@ def validate_partition(partition: NodeList) -> bool:
                 # external user node, need to expose as an output
                 outputs.append(user_node)
 
-    # perform DFS on the parition outputs
-    # if it reaches a node within the partition, then it found a cycle
-    visited: NodeSet = set()
-
-    def dfs_find_cycle(node):
-        if node in partition_set:
-            return True  # found cycle, return
-
-        visited.add(node)
-        for user_node in node.users:
-            if user_node not in visited:
-                if dfs_find_cycle(user_node):
-                    return True
+    # Perform DFS on the partition outputs.
+    # If it reaches a node within the partition, then it found a cycle.
+    def dfs_find_cycle(node, visited):
+        # Start with `node` and traverse through (toward child nodes)
+        # its connected sub-graph. Nodes in `visited` won't be added
+        # to `queue` again.
+        queue: NodeList = [node]
```

Comment:
nit, you can pass all the output nodes into this function in one shot, and initialize the queue with the list of nodes.